### PR TITLE
PXC-3528: PS 8.0.22 merge

### PR DESCRIPTION
**Pull request by kamil-holubicki**
https://jira.percona.com/browse/PXC-3528

Merge of PS 8.0.22 into PXC. For details see the particular commits messages.

### CODE REVIEW

File: share/messages_to_clients.txt [hunk 1]
```diff
@@ -9552,7 +9552,7 @@ ER_DTE_ENCRYPTION_THREADS_ACTIVE
 ER_ENCRYPTION_KEY_ID_VALUE_OUT_OF_RANGE
   eng "The requested value %lu for the encryption_key_id exceeds the maximum allowed %lu"
 
-ER_WRAPPED_UDF_EXCEPTION
+ER_WRAPPED_UDF_EXCEPTION
```
Comment (dutow) [inline share/messages_to_clients.txt]: Whitespace change?

File: cmake/maintainer.cmake [hunk 2]
```diff
@@ -67,7 +67,15 @@ IF(MY_COMPILER_IS_GNU)
...
   IF(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
     # GCC 8 has bugs with "final".
-    MY_ADD_CXX_WARNING_FLAG("Wsuggest-override")
+
+    # Disable this flag for PXC. Codership's wsrep-lib part and wsrep server
+    # hooks are not compatible with this flag. This causes errors during
+    # compilation of server part and storage engines including wsrep-lib and 
+    # wsrep hooks headers.
+    # Enable it back when wsrep-lib part is fixed. Disabled for now instead of
+    # fixing wsrep-lib to make next upstream merges easier.
```
Comment (dutow) [inline cmake/maintainer.cmake:76]: Maybe we should create a ticket about this? As I'm not sure when upstream will fix this.

File: mysql-test/collections/disabled.def [hunk 3]
```diff
@@ -37,10 +37,37 @@ innodb.innodb_bug14676111                         : BUG#18200831 The test has ne
...
+innodb.innodb_page_size_func                      : BUG#0 PXC requires 16k InnoDB page size
+innodb.innodb_pagesize_max_recordsize             : BUG#0 PXC requires 16k InnoDB page size
+innodb.mysqldump_max_recordsize                   : BUG#0 PXC requires 16k InnoDB page size
+innodb.blob-crash-4k                              : BUG#0 PXC requires 16k InnoDB page size
 
 # innodb_fts suite test
 innodb_fts.sync_block : BUG#2232 percona https://jira.percona.com/browse/PS-2232
 
+# innodb_zip
+innodb_zip.4k                                     : BUG#0 PXC requires 16k InnoDB page size
```
Comment (dutow) [inline mysql-test/collections/disabled.def]: This is PXC-3129

File: mysql-test/collections/disabled.def [hunk 4]
```diff
@@ -251,6 +283,67 @@ galera.galera_fk_lock_parent_update_child : BUG#0 Few cases have timing issues(P
 galera.galera_toi_ddl_fk_insert : BUG#0 CODERSHIP qa#39 test fails sporadically (PXC-3431)
 
 
+# encryption
+# Probably we should use upstream version of encryption tests for PXC
+encryption.percona_parallel_dblwr_encrypt                               : BUG#0 PXC does not support KEYRING and ONLINE_TO_KEYRING encyption methods
```
Comment (dutow) [inline mysql-test/collections/disabled.def:289]: It should be safe to remove the encryption testsuite from the defaults for pxc.

File: share/messages_to_clients.txt (hunk 1 above)
Comment (kamil-holubicki) [reply share/messages_to_clients.txt]: fixed

**Comment by venkatesh-prasad-v** [comment]
As of now, `wsrep_sst_xtrabackup` script requires PXB version 8.0.13. Since PXC-8.0.22 is incompatible with 8.0.13, please update `scripts/wsrep_sst_xtrabackup-v2.sh` to set the minimum version to PXB-8.0.22-15.

Also, please set the correct PXB tag in `cmake/dev-helper.cmake`.

File: cmake/maintainer.cmake (hunk 2 above)
Comment (kamil-holubicki) [reply cmake/maintainer.cmake:76]: PXC-3556

File: mysql-test/collections/disabled.def (hunk 4 above)
Comment (kamil-holubicki) [reply mysql-test/collections/disabled.def:289]: PXC-3557

File: mysql-test/collections/disabled.def (hunk 3 above)
Comment (kamil-holubicki) [reply mysql-test/collections/disabled.def]: fixed

**Comment by kamil-holubicki** [comment]
@venkatesh-prasad-v We package the newest PXB binaries together with PXC, so they are always aligned, however it is good to keep it consistent in scripts as well. Fixed.

File: mysql-test/suite/perfschema/t/error_log.test [hunk 5]
```diff
@@ -252,7 +252,7 @@ UNINSTALL COMPONENT "file://component_log_sink_json";
  
 --echo
 --echo # Restarting server with --log-error-verbosity=1
---let restart_parameters="restart: --log-error-verbosity=1"
+--let restart_parameters="restart: --log-error-verbosity=1 --wsrep-min-log-verbosity=1"
```
Comment (venkatesh-prasad-v) [inline mysql-test/suite/perfschema/t/error_log.test:255]: Not a strong opinion. Wouldn't it be better to pass `--wsrep-min-log-verbosity=1` option through an opt file so that the test uses same option for restarting every time?

File: mysql-test/t/all_persisted_variables.test [hunk 6]
```diff
@@ -144,7 +153,9 @@ ORDER BY variable_name;
 while($var_id <= $count_vars)
 {
   --let $var_names= `SELECT var_name FROM all_vars WHERE id=$var_id;`
-  --error 0,ER_INCORRECT_GLOBAL_LOCAL_VAR,ER_WRONG_VALUE_FOR_VAR
+  --echo "variable: $var_names"
```
Comment (venkatesh-prasad-v) [inline mysql-test/t/all_persisted_variables.test]: Is printing all variables intentional? Or is it a debug print?

File: mysql-test/t/all_persisted_variables.test [hunk 7]
```diff
@@ -73,7 +76,10 @@ performance_schema.global_variables WHERE variable_name NOT IN
 'innodb_monitor_reset',
 'innodb_monitor_reset_all',
 'innodb_default_encryption_key_id',
-'rbr_exec_mode');
+'rbr_exec_mode',
+'pxc_maint_mode',
+'pxc_strict_mode',
+'wsrep_trx_fragment_size');
```
Comment (venkatesh-prasad-v) [inline mysql-test/t/all_persisted_variables.test:82]: Why this change was required?

**Review by dutow — APPROVED** [review]

File: mysql-test/t/all_persisted_variables.test (hunk 6 above)
Comment (kamil-holubicki) [reply mysql-test/t/all_persisted_variables.test]: Yup, leftover after debugging. 
Fixed.

File: mysql-test/t/all_persisted_variables.test (hunk 7 above)
Comment (kamil-holubicki) [reply mysql-test/t/all_persisted_variables.test:82]: `Query 'SET GLOBAL $var_name = @@global.$var_name' failed with wrong error 1105: 'pxc_maint_mode can be changed only if node is cluster-node'`
1105 is ER_UNKNOWN_ERROR. I don't want to add it to the list of accepted errors as it could suppress too much and also don't want to change current software behavior.

Similar situation for two other variables

`Query 'SET GLOBAL $var_name = @@global.$var_name' failed with wrong error 1105: 'pxc_strict_mode can be changed only if node is cluster-node',`
